lib/enter.js

```diff
@@ -1,34 +1,43 @@
-var partialRight = require('./partial-right')
-  , d3 = require('d3')
-  , DnD = require('./dnd')
+var DnD = require('./dnd')
+  , d3 = require('d3-selection')
+  , drag = require('d3-drag').drag
+  , partialRight = require('./partial-right')
 
 module.exports = function (tree) {
   var dnd = new DnD(tree)
-    , listener = d3.behavior.drag()
+    , listener = drag()
 
-  listener.on('dragstart', dnd.start)
+  listener.on('start', dnd.start)
           .on('drag', dnd.drag)
-          .on('dragend', dnd.end)
+          .on('end', dnd.end)
+          .subject(function () {
+            // This is a weird addition from upgrading d3 v3 to v4. The new d3-drag functionality sets a subect
+            // that uses d.y and d.x. Since we're setting x/y values on our data, this breaks. We don't want to use
+            // the d.x d.y to find offsets. So this overrides subject, so it uses the event x/y
+            return {
```

Comment:
Great find. This is a good point. Since we're now generating a layout ourselves, d3 isn't setting x/y values on the object, so I may be able to remove this subject callback. Checking now.